### ISSUE TITLE
Connection issues while using on databricks

### ISSUE THREAD

**Issue by Harshitg** [microsoft/sql-spark-connector#9]
Any idea how to fix this?

`: java.lang.ClassNotFoundException: Failed to find data source: com.microsoft.sqlserver.jdbc.spark. Please find packages at http://spark.apache.org/third-party-projects.html`

```
conf = SparkConf() \
    .setAppName(appName) \
    .setMaster(master) \
    .set("spark.driver.extraClassPath","C:/Users/XXXX//mssql-jdbc-8.3.1.jre14-preview.jar")\
    .set("spark.sql.execution.arrow.enabled", True)

```

**Comment by shivsood** [comment]
@Harshitg  This indicates jar not avialble for Spark to load.

**Comment by Harshitg** [comment]
> @Harshitg This indicates jar not avialble for Spark to load. On Azure Databricks i would generally load by adding the jar as a library for the while cluster to use.

Thanks Shiv, would you be able to tell me the name of the Jar file to be loaded. I have tried using `apache-spark-sql-connector.jar` and  `mssql-jdbc-8.3.1.jre14-preview.jar` with similar results.

**Comment by mokabiru** [comment]
@Harshitg You don't need to install the default JDBC connector as the Microsoft SQL Server JDBC driver library (mssql-jdbc) is pre-installed in Databricks Runtime 3.4 and above, the new Spark connector for SQL needs to be exclusively installed using the .JAR file. Which version of Databricks Runtime and Python version are you using?

**Comment by Harshitg** [comment]
I shifted to local to try this with possible combinations, the outcome is as follows:

Spark: 3.0.0
Environment: Windows

Code:
```
from pyspark import SparkContext, SparkConf, SQLContext

appName = "PySpark SQL Server"
master = "local[*]"
conf = SparkConf() \
    .setAppName(appName) \
    .setMaster(master) \
    .set("spark.driver.extraClassPath","c:/users/xxxx/mssql-jdbc-8.2.1.jre8.jar") \
    .set("spark.driver.extraClassPath","C:/Users/xxxx/apache-spark-sql-connector.jar") \
    .set("spark.sql.execution.arrow.enabled", True)
sc = SparkContext.getOrCreate(conf=conf)
sqlContext = SQLContext(sc)
spark = sqlContext.sparkSession

hostname = "hostname"
database = "database"
port = "port"
table = "table"
user = "user"
password  = "password"

# The code as recommended in readme
jdbcDF = spark.read \
        .format("com.microsoft.sqlserver.jdbc.spark") \
        .option("url", f"jdbc:sqlserver://{hostname}:{port};databaseName={database}") \
        .option("dbtable", table) \
        .option("user", user) \
        .option("password", password).load()
```
```
# The above code leads to following exception
 An error occurred while calling o75.load.
: java.lang.NoClassDefFoundError: org/apache/spark/internal/Logging$class
	at com.microsoft.sqlserver.jdbc.spark.DefaultSource.<init>(DefaultSource.scala:19)
```
```
# Tried a combination to check
jdbcDF = spark.read \
        .format("com.microsoft.sqlserver.jdbc.spark") \
        .option("url", f"jdbc:sqlserver://{hostname}:{port};databaseName={database}") \
        .option('driver', 'com.microsoft.sqlserver.jdbc.SQLServerDriver') \
        .option("dbtable", table) \
        .option("user", user) \
        .option("password", password).load()
```
```
# The above code leads to following exception
An error occurred while calling o85.load.
: java.lang.NoClassDefFoundError: org/apache/spark/internal/Logging$class
	at com.microsoft.sqlserver.jdbc.spark.DefaultSource.<init>(DefaultSource.scala:19)
```
```
# The code that we generally use in our environment
jdbcDF = spark.read.format("jdbc") \
    .option("url", f"jdbc:sqlserver://{hostname}:{port};databaseName={database}") \
    .option("dbtable", table) \
    .option("user", user) \
    .option("password", password) \
    .option('driver', 'com.microsoft.sqlserver.jdbc.SQLServerDriver')\
    .load()
```
This works fine.

**Comment by shivsood** [comment]
Ack on the issue. Have not tested/fixed on Spark 3.0.

**Comment by rajmera3** [comment]
Thanks @Harshitg 

I've created a new issue for Spark 3.0 support and am closing this issue since this is not a databricks connection issue per say.
